uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsencePeriod.php

```diff
@@ -104,6 +106,34 @@ private static function validateDates($params) {
   }
 
   /**
+   * Checks if another absence period exists with same title as
+   * the absence period being created/updated and throws an exception if found.
+   *
+   * @param array $params
+   *
+   * @throws \CRM_HRLeaveAndAbsences_Exception_InvalidAbsencePeriodException
+   */
+  private static function validateAbsencePeriodTitle($params) {
+    $title = !empty($params['title']) ? $params['title'] : '';
+    $absencePeriod = new self();
+    $absencePeriod->title = $title;
+    $absencePeriod->find(true);
+
+    if (!$absencePeriod->id) {
+      return;
+    }
+
+    $throwExceptionOnCreate = empty($params['id']);
+    $throwExceptionOnUpdate = !empty($params['id']) && $absencePeriod->id != $params['id'];
+
+    if ($throwExceptionOnCreate || $throwExceptionOnUpdate) {
+      throw new InvalidAbsencePeriodException(
+        'Absence Period with same title already exists!'
```

Comment:
I agree but looking at the other exceptions we throw in L&A, we don't usually do this, so I'll just leave it like this to keep it consistent.